playground/tei_bot.py

```diff
@@ -0,0 +1,155 @@
+import numpy as np
+from typing import List
+from board import Stone
+
+
+class TeiBot:
+
+    def __init__(self, log: List[Stone]):
+        self.log = log
+
+    @staticmethod
+    def array_to_board(x: int) -> str:
+        if x == 0:
+            points = "+"
+        elif x == 1:
+            points = "B"
+        elif x == 2:
+            points = "W"
+        else:  # x == -1:
+            points = "S"
+        return points
+
+    @staticmethod
+    def stone_to_array(log: List[Stone]) -> np.array:
+        stone_array = np.zeros((19, 19))
+        for pos in log:
+            x = int(pos.x) - 1
+            y = int(pos.y) - 1
+            if pos.color == "b":
+                stone_array[y, x] = 1
+            elif pos.color == "w":
+                stone_array[y, x] = 2
+            elif pos.color == "s":
+                stone_array[y, x] = -1
+        return stone_array
+
+    @staticmethod
+    def array_to_stone(stone_array: np.array, color: str) -> List[Stone]:
+        if color == "b":
+            out = 1
+        elif color == "w":
+            out = 2
+        else:
+            out = -1
+
+        y, x = np.where(stone_array == out)
+        positions = list(zip(list(x), list(y)))
+        stones = [Stone(str(x + 1), str(y + 1), color) for x, y in positions]
+
+        return stones
+
+    @staticmethod
+    def move_stone(stone, direction, stride=1):
+        x = int(stone.x)
+        y = int(stone.y)
+
+        if direction == 'l':
+            x = x - stride
+        elif direction == 'r':
+            x = x + stride
+        elif direction == 'u':
+            y = y - stride
+        elif direction == 'd':
+            y = y + stride
+        elif direction == 'ul':
+            y = y - stride
+            x = x - stride
+        elif direction == 'ur':
+            y = y - stride
+            x = x + stride
+        elif direction == 'dl':
+            y = y + stride
+            x = x - stride
+        elif direction == 'dr':
+            y = y + stride
+            x = x + stride
+
+        return Stone(str(x), str(y), stone.color)
+
+    def draw_board(self):
+        stone_array = self.stone_to_array(self.log)
+        join_column = list(map(lambda y: " ".join([self.array_to_board(x) for i, x in enumerate(y)]), stone_array))
+        join_row = "\n" + "\n".join(join_column)
+        return join_row
+
+    def suggest_position(self, lower=1, upper=19):
```

Comment:
since the type of `position` is in list(`[]`), `suggest_position` could be renamed as `suggest_positions`

Reply:
need return type